plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js

```diff
@@ -193,7 +193,10 @@ export default {
               this.chatService = chatService;
 
               if (this.oneOnOneMessage) {
-                this.channel.chatable.users[0].trackStatus();
+                const user = this.channel.chatable.users[0];
+                if (user.username !== I18n.t("chat.deleted_chat_username")) {
```

Comment:
This seems kinda risky... because it will be triggered if someone changes their username to 'deleted'.

Could we have some kind of `deleted: true` property on the chatable user, so that we can do this more safely?

Reply:
It's a bit risky yes, but it's not a part of this fix, I'm just using what was implemented before in https://github.com/discourse/discourse-chat/pull/1149 (speaking of that fix, by the way, even though it's risky, it makes a lot of things simpler, and I feel that implementation can be refactored to using `deleted:true`).

Reply:
That implementation declares the same string on the server side, here I needed the same string on the client. I understand this can regress, but in that case we'll just start seeing the warning in the console, which is not critical.